internal/controller/workspacetemplate_controller.go

```diff
@@ -141,7 +165,8 @@ func (r *WorkspaceTemplateReconciler) handleDeletion(ctx context.Context, templa
 	}
 
 	// Check if any workspaces are using this template
-	workspaces, err := r.templateResolver.ListWorkspacesUsingTemplate(ctx, template.Name)
+	// Use empty namespace to match all workspaces (backwards compatible)
+	workspaces, _, err := workspace.ListByTemplate(ctx, r.Client, template.Name, "", "", 0)
```

Comment:
any way to specify a `limit=1`, ie return at most 1 results? That should be more efficient given the user case.